### DISCUSSION TITLE
Homebrew tap for neovim nightly binaries 🍺

### DISCUSSION THREAD

**Discussion by benjiwolff** [neovim/neovim#28146] [Show and tell]
In #28125 I was confused to find that `brew install neovim --HEAD` (see [Homebrew formula](https://formulae.brew.sh/formula/neovim)) does not install all dependencies necessary to run the development/nightly version of neovim, so I made a homebrew cask tap that downloads the nightly build. This should make it a bit more convenient to install and update neovim nightly. Feel free to fork this onto [neovim](https://github.com/neovim).

[benjiwolff/homebrew-neovim-nightly](https://github.com/benjiwolff/homebrew-neovim-nightly)

**Comment by clason** [comment]
No thank you; we're not interested in maintaining distro-specific builds, and we're adamant that if you use nightly, you should get it straight from our repo -- after all, you're testing Neovim, not distro builds or installations.

**Reply by benjiwolff** [reply]
I understand. However, I find these 20 lines of code comparable to the release notes in terms of maintenance since they are already specific to macOS. For that little effort, it seems a bit weird to me, not to support Homebrew. Especially, if the goal is that people stay up to date with the latest nightly releases.

**Reply by clason** [reply]
It's the principle.
1. We don't support _any_ distros, period (else where to draw the line?)
2. We don't support using nightlies for production use (they can break too often); you're welcome to, of course, but if you're using one, you're implicitly agreeing to help with Neovim development. And adding indirection makes this harder (for testing and bisecting purposes).

**Reply by benjiwolff** [reply]
1. You _do_ explicitly support macOS. It's not like there are 20 macOS distributions and package managers. Linux is another story and I am not saying that the same effort/reward tradeoff applies there.
2. Using a package manager does *not* imply that I am using the nightly release in production or that I do not agree with helping in the development of Neovim. After all, I still have to opt into the nightly version. For this reason I do *not* suggest it is added to the default homebrew-cask tap but rather to a standalone neovim repository.

**Reply by clason** [reply]
1. We do support macOS _as a platform_, but the only _distribution_ we support (on macOS and elsewhere) is a) our own releases, from our own Releases page and b) build from source, with our build scripts.

2. Helping with development implies testing with a known installation that _we_ control.

**Comment by dundargoc** [comment]
Can you not just send a PR to homebrew instead to fix that?

**Reply by clason** [reply]
No, because that (no parsers) is intentional. Homebrew does not support `--HEAD` builds either.

**Comment by unphased** [comment]
I would like to mention that nvim HEAD build fails with homebrew for me in this way: 

```
==> Downloading from https://codeload.github.com/tree-sitter-grammars/tree-sitter-markdown/tar.gz/refs/tags/v0.3.2
################################################################################################################################################################### 100.0%
==> cmake -S /private/tmp/neovim-20250215-69222-jhwd24/deps-build/build/src/treesitter-markdown -B /private/tmp/neovim-20250215-69222-jhwd24/deps-build/build/treesitter-m
==> cmake --build /private/tmp/neovim-20250215-69222-jhwd24/deps-build/build/treesitter-markdown
==> cmake --install /private/tmp/neovim-20250215-69222-jhwd24/deps-build/build/treesitter-markdown
==> cmake -S . -B build -DLUV_LIBRARY=/opt/homebrew/opt/luv/lib/libluv.dylib -DLIBUV_LIBRARY=/opt/homebrew/opt/libuv/lib/libuv.dylib -DLPEG_LIBRARY=/opt/homebrew/opt/lpeg
==> cmake --build build
Last 15 lines from /Users/slu/Library/Logs/Homebrew/neovim/20.cmake:
 1135 | uint32_t ts_language_version(const TSLanguage *self);
      |          ^
/tmp/neovim-20250215-69222-jhwd24/src/nvim/lua/treesitter.c:303:22: error: call to undeclared function 'ts_language_abi_version'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
  303 |   lua_pushinteger(L, ts_language_abi_version(lang));  // [retval, version]
      |                      ^
2 errors generated.
[ 80%] Building C object src/nvim/CMakeFiles/nvim_bin.dir/mark.c.o
cd /tmp/neovim-20250215-69222-jhwd24/build/src/nvim && /opt/homebrew/Library/Homebrew/shims/mac/super/clang -DINCLUDE_GENERATED_DECLARATIONS -DUNIT_TESTING -DUTF8PROC_STATIC -D_GNU_SOURCE -Dnvim_bin_EXPORTS -I/tmp/neovim-20250215-69222-jhwd24/build/src/nvim/auto -I/tmp/neovim-20250215-69222-jhwd24/build/include -I/tmp/neovim-20250215-69222-jhwd24/build/cmake.config -I/tmp/neovim-20250215-69222-jhwd24/src -isystem /opt/homebrew/include/luajit-2.1 -isystem /opt/homebrew/opt/gettext/include -O3 -DNDEBUG -flto=thin -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk -mmacosx-version-min=15.2 -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wvla -Wdouble-promotion -Wmissing-noreturn -Wmissing-format-attribute -Wmissing-prototypes -fsigned-char -fstack-protector-strong -Wimplicit-fallthrough -fdiagnostics-color=auto -MD -MT src/nvim/CMakeFiles/nvim_bin.dir/mark.c.o -MF CMakeFiles/nvim_bin.dir/mark.c.o.d -o CMakeFiles/nvim_bin.dir/mark.c.o -c /tmp/neovim-20250215-69222-jhwd24/src/nvim/mark.c
make[2]: *** [src/nvim/CMakeFiles/nvim_bin.dir/lua/treesitter.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
[ 80%] Building C object src/nvim/CMakeFiles/nvim_bin.dir/marktree.c.o
cd /tmp/neovim-20250215-69222-jhwd24/build/src/nvim && /opt/homebrew/Library/Homebrew/shims/mac/super/clang -DINCLUDE_GENERATED_DECLARATIONS -DUNIT_TESTING -DUTF8PROC_STATIC -D_GNU_SOURCE -Dnvim_bin_EXPORTS -I/tmp/neovim-20250215-69222-jhwd24/build/src/nvim/auto -I/tmp/neovim-20250215-69222-jhwd24/build/include -I/tmp/neovim-20250215-69222-jhwd24/build/cmake.config -I/tmp/neovim-20250215-69222-jhwd24/src -isystem /opt/homebrew/include/luajit-2.1 -isystem /opt/homebrew/opt/gettext/include -O3 -DNDEBUG -flto=thin -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk -mmacosx-version-min=15.2 -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wvla -Wdouble-promotion -Wmissing-noreturn -Wmissing-format-attribute -Wmissing-prototypes -fsigned-char -fstack-protector-strong -Wimplicit-fallthrough -fdiagnostics-color=auto -MD -MT src/nvim/CMakeFiles/nvim_bin.dir/marktree.c.o -MF CMakeFiles/nvim_bin.dir/marktree.c.o.d -o CMakeFiles/nvim_bin.dir/marktree.c.o -c /tmp/neovim-20250215-69222-jhwd24/src/nvim/marktree.c
make[1]: *** [src/nvim/CMakeFiles/nvim_bin.dir/all] Error 2
make: *** [all] Error 2
```

And that your cask has helped me get on with my life. 

I have created this separate discussion: https://github.com/neovim/neovim/discussions/32450 to ask for clarification from the nvim project about how to consume its releases, which I understand from the above discussion are the only supported nightly releases.
